presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -556,27 +552,13 @@ private void invokeNoMoreSplitsIfNecessary()
             InternalHiveSplitFactory splitFactory,
             boolean splittable)
     {
-        // Strategy: List once, filter file listing based on manifest file.
-
         List<Iterator<LocatedFileStatus>> fileStatusIterators = new ArrayList<>();
         List<Path> paths = new ArrayList<>();
 
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(fileSystem.open(symlinkFile.getPath()), StandardCharsets.UTF_8))) {
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(
+                hdfsEnvironment.doAs(hdfsContext.getIdentity().getUser(), () -> fileSystem.open(symlinkFile.getPath())), StandardCharsets.UTF_8))) {
```

Comment:
what about obtaining `targetFilesystem`? Does it need `doAs` too?

Reply:
HdfsEnvironment.getFileSystem contains a doAs https://github.com/prestosql/presto/blob/master/presto-hive/src/main/java/io/prestosql/plugin/hive/HdfsEnvironment.java#L76